requirements.txt

```
@@ -2,6 +2,6 @@ numpy>=1.16.5
 scipy>=1.3.0
 pandas>=1.2.0
 scikit-learn>=0.21.2
-sqlalchemy>=1.3.6
+sqlalchemy==1.4.42
```

```suggestion
sqlalchemy<=1.4.42
```

Do you think it makes sense to be less restrictive? If you disagree, I will merge with a pinned version

---

I'd recommend going with `sqlalchemy>=1.3.6,<2`, or `sqlalchemy~=1.4.42` if you don't want an exact pin - I assume the previous version minimum was for a reason. But your call.

---

@d33bs - what do you recommend?

---

Thanks @gwaybio - I would recommend what @bethac07 mentioned with a version range here, respecting that any broken aspects of the SQLAlchemy API in v2 may incur separate updates for a time. Specifically, I'd vote for `sqlalchemy>=1.3.6,<2` to help make sure we account for any reasons why 1.3.6 was preferred at one point or another. Dependencies could be updated and tested otherwise (for ex. updating >= 1.4.42) outside of the focus of this PR and #250 .

---

Cool, have updated to @d33bs 's preference, since already approved will pull when tests are done!